File: 200-number-of-islands/200-number-of-islands.py
def dfs(grid, row, col):
        if row < 0 or col < 0 or row >= len(grid) or col >= len(grid[0]) or grid[row][col] == '0':
            return 
        grid[row][col] = '0'
        dfs(grid, row+1, col)
        dfs(grid, row-1, col)
        dfs(grid, row, col+1)
        dfs(grid, row, col-1)

class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        if not grid:
            return 0 
        count = 0
        for row in range(len(grid)):
            for col in range(len(grid[0])):
                if grid[row][col] == '1':
                    dfs(grid, row, col)
                    count += 1
        return count 

   

File: 200-number-of-islands/200-number-of-islands.cpp
class Solution {
public:
    void dfs(int i, int j, int row, int col, vector<vector<char>> &grid){
        if(i<0 || i>=row || j<0 || j>=col || grid[i][j]=='0') return;
        grid[i][j] = '0';
        dfs(i-1, j, row, col, grid);
        dfs(i+1, j, row, col, grid);
        dfs(i, j-1, row, col, grid);
        dfs(i, j+1, row, col, grid);
    }
    
    int numIslands(vector<vector<char>>& grid) {
        int count = 0;
        int row = grid.size();
        int col = grid[0].size();
        for(int i=0; i<row; i++){
            for(int j=0; j<col; j++){
                if(grid[i][j]=='1'){
                    count += 1;
                    dfs(i, j, row, col, grid);
                }
            }
        }
        return count;
        
    }
};